src/txn/txn_rollback_to_stable.c

```diff
@@ -464,7 +464,8 @@ __rollback_abort_row_reconciled_page_internal(WT_SESSION_IMPL *session, const vo
         image_local = tmp.data;
     }
 
-    page_flags = WT_DATA_IN_ITEM(&tmp) ? WT_PAGE_DISK_ALLOC : WT_PAGE_DISK_MAPPED;
+    /* Don't free the passed image later. */
+    page_flags = image != NULL ? 0 : WT_PAGE_DISK_ALLOC;
```

Comment:
This has already been merged, but it doesn't "feel" right to me.

The WT_PAGE_DISK_ALLOC flag controls whether the memory allocated by page_inmem is counted against the cache as well as whether it's been allocated externally. Why shouldn't it be in this case? i.e: why would this pass a `0` to `__wt_page_inmem`?

It seems as though maybe that happens in other cases like salvage as well. @kommiharibabu you suggested this fix. Could you open a ticket to review cases where page in memory is called without an allocation option - I think we might need a new explicit flag here to cover the cases currently using 0.

Reply:
Understood. Created WT-5949.